### PR TITLE
fileutil: rework purge tests so they don't poll

### LINKED ISSUE #5966: TestPurgeFileHoldingLockFile fails in semaphore
```
2016-07-17 06:59:58.489255 I | fileutil: purged file /tmp/purgefile255810807/0.test successfully
2016-07-17 06:59:58.489421 I | fileutil: purged file /tmp/purgefile255810807/1.test successfully
2016-07-17 06:59:58.489479 I | fileutil: purged file /tmp/purgefile255810807/2.test successfully
2016-07-17 06:59:58.502014 I | fileutil: purged file /tmp/purgefile255810807/3.test successfully
2016-07-17 06:59:58.516947 I | fileutil: purged file /tmp/purgefile255810807/4.test successfully
2016-07-17 06:59:58.526532 I | fileutil: purged file /tmp/purgefile255810807/5.test successfully
2016-07-17 06:59:58.541437 I | fileutil: purged file /tmp/purgefile255810807/6.test successfully
2016-07-17 06:59:58.550338 I | fileutil: purged file /tmp/purgefile439700458/0.test successfully
2016-07-17 06:59:58.550446 I | fileutil: purged file /tmp/purgefile439700458/1.test successfully
2016-07-17 06:59:58.550502 I | fileutil: purged file /tmp/purgefile439700458/2.test successfully
2016-07-17 06:59:58.550553 I | fileutil: purged file /tmp/purgefile439700458/3.test successfully
2016-07-17 06:59:58.550602 I | fileutil: purged file /tmp/purgefile439700458/4.test successfully
2016-07-17 06:59:58.566723 I | fileutil: purged file /tmp/purgefile439700458/5.test successfully
2016-07-17 06:59:58.566804 I | fileutil: purged file /tmp/purgefile439700458/6.test successfully
2016-07-17 06:59:58.678471 I | fileutil: purged file /tmp/purgefile242953684/0.test successfully
2016-07-17 06:59:58.678583 I | fileutil: purged file /tmp/purgefile242953684/1.test successfully
2016-07-17 06:59:58.678643 I | fileutil: purged file /tmp/purgefile242953684/2.test successfully
2016-07-17 06:59:58.689115 I | fileutil: purged file /tmp/purgefile242953684/3.test successfully
2016-07-17 06:59:58.699337 I | fileutil: purged file /tmp/purgefile242953684/4.test successfully
2016-07-17 06:59:58.709703 I | fileutil: purged file /tmp/purgefile242953684/5.test successfully
2016-07-17 06:59:58.723438 I | fileutil: purged file /tmp/purgefile242953684/6.test successfully
2016-07-17 06:59:58.734472 I | fileutil: purged file /tmp/purgefile259074851/0.test successfully
2016-07-17 06:59:58.734582 I | fileutil: purged file /tmp/purgefile259074851/1.test successfully
2016-07-17 06:59:58.734646 I | fileutil: purged file /tmp/purgefile259074851/2.test successfully
2016-07-17 06:59:58.734704 I | fileutil: purged file /tmp/purgefile259074851/3.test successfully
2016-07-17 06:59:58.734761 I | fileutil: purged file /tmp/purgefile259074851/4.test successfully
2016-07-17 06:59:58.745817 I | fileutil: purged file /tmp/purgefile259074851/5.test successfully
2016-07-17 06:59:58.745906 I | fileutil: purged file /tmp/purgefile259074851/6.test successfully
2016-07-17 06:59:58.863553 I | fileutil: purged file /tmp/purgefile523444373/0.test successfully
2016-07-17 06:59:58.863665 I | fileutil: purged file /tmp/purgefile523444373/1.test successfully
2016-07-17 06:59:58.863730 I | fileutil: purged file /tmp/purgefile523444373/2.test successfully
2016-07-17 06:59:58.883533 I | fileutil: purged file /tmp/purgefile523444373/3.test successfully
2016-07-17 06:59:58.916460 I | fileutil: purged file /tmp/purgefile523444373/4.test successfully
2016-07-17 06:59:58.925566 I | fileutil: purged file /tmp/purgefile523444373/5.test successfully
2016-07-17 06:59:58.936863 I | fileutil: purged file /tmp/purgefile523444373/6.test successfully
2016-07-17 06:59:58.947136 I | fileutil: purged file /tmp/purgefile977304816/0.test successfully
2016-07-17 06:59:59.070026 I | fileutil: purged file /tmp/purgefile977304816/1.test successfully
2016-07-17 06:59:59.070117 I | fileutil: purged file /tmp/purgefile977304816/2.test successfully
2016-07-17 06:59:59.070187 I | fileutil: purged file /tmp/purgefile977304816/3.test successfully
2016-07-17 06:59:59.070253 I | fileutil: purged file /tmp/purgefile977304816/4.test successfully
2016-07-17 06:59:59.070317 I | fileutil: purged file /tmp/purgefile977304816/5.test successfully
2016-07-17 06:59:59.070383 I | fileutil: purged file /tmp/purgefile977304816/6.test successfully
--- FAIL: TestPurgeFileHoldingLockFile (0.13s)
	purge_test.go:126: filenames = [1.test 2.test 3.test 4.test 5.test 6.test 7.test 8.test 9.test], want [5.test 6.test 7.test 8.test 9.test]
FAIL
coverage: 58.7% of statements
```

### PR DESCRIPTION
**Pull request by heyitsanthony**
Fixes #5966

### CODE REVIEW

File: pkg/fileutil/purge.go [hunk 1]
```diff
@@ -23,6 +23,10 @@ import (
 )
 
 func PurgeFile(dirname string, suffix string, max uint, interval time.Duration, stop <-chan struct{}) <-chan error {
+	return purgeFile(dirname, suffix, max, interval, stop, nil)
+}
+
+func purgeFile(dirname string, suffix string, max uint, interval time.Duration, stop <-chan struct{}, purgec chan<- string) <-chan error {
```
Comment (gyuho) [inline pkg/fileutil/purge.go]: So `purgec` is only for testing?

**Comment by gyuho** [comment]
lgtm. defer to @xiang90

File: pkg/fileutil/purge.go (hunk 1 above)
Comment (heyitsanthony) [inline pkg/fileutil/purge.go]: yes

Comment (xiang90) [inline pkg/fileutil/purge.go]: probably add a comments about the purgec thing?

**Comment by xiang90** [comment]
lgtm after documenting purgec chan.

**Comment by heyitsanthony** [comment]
only failing on `TestEmbed`; merging

**Pull request merged**